microsoft-edge/webview2/concepts/print.md

```diff
@@ -0,0 +1,736 @@
+---
+title: Printing in WebView2 apps
+description: How to print a web page in a WebView2 app.
+author: MSEdgeTeam
+ms.author: msedgedevrel
+ms.topic: conceptual
+ms.prod: microsoft-edge
+ms.technology: webview
+ms.date: 01/12/2023
+---
+# Print
+
+There are a few different ways to print a web page in WebView2 which give you varying levels of ease of use and control.
+
+## ShowPrintUI
+
+The ShowPrintUI show the WebView2 print preview dialog or OS print dialog for the current top-level document in the WebView2. With this mechanism you easily get a printing experience the user will be familiar with.
+
+<!-- ------------------------------ -->
+
+# [C#](#tab/csharp)
+
+[CoreWebView2.ShowPrintUI() Method](/dotnet/api/microsoft.web.webview2.core.corewebview2.showprintui#microsoft-web-webview2-core-corewebview2-showprintui)
+[CoreWebView2.ShowPrintUI(CoreWebView2PrintDialogKind) Method](/dotnet/api/microsoft.web.webview2.core.corewebview2.showprintui#microsoft-web-webview2-core-corewebview2-showprintui(microsoft-web-webview2-core-corewebview2printdialogkind))
+[CoreWebView2PrintDialogKind Enum](/dotnet/api/microsoft.web.webview2.core.corewebview2printdialogkind)
+
+
+<!-- ------------------------------ -->
+
+# [C++](#tab/cpp)
+
+[ICoreWebView2_16::ShowPrintUI(printDialogKind) Method](/microsoft-edge/webview2/reference/win32/icorewebview2_16#showprintui)
+[COREWEBVIEW2_PRINT_DIALOG_KIND Enum](/microsoft-edge/webview2/reference/win32/icorewebview2_16#corewebview2_print_dialog_kind)
+
+
+---
+
+<!-- end of tab-set -->
+
+### Example: ShowPrintUI
+
+This example shows the user a print dialog. If `printDialog` is `CoreWebView2PrintDialogKind.Browser`, opens a browser print preview dialog, `CoreWebView2PrintDialogKind.System` opens a system print dialog.
+
+
+<!-- ------------------------------ -->
+
+# [C#](#tab/csharp)
+
+```csharp
+void ShowPrintUI(object target, ExecutedRoutedEventArgs e)
+{
+  string printDialog = e.Parameter.ToString();
+  if (printDialog == "Browser")
+  {
+    // Opens the browser print preview dialog.
+    webView.CoreWebView2.ShowPrintUI();
+  }
+  else
+  {
+    // Opens the system print dialog.
+    webView.CoreWebView2.ShowPrintUI(CoreWebView2PrintDialogKind.System);
+  }
+}
+```
+
+**APIs:**
+
+* [CoreWebView2.ShowPrintUI() Method](/dotnet/api/microsoft.web.webview2.core.corewebview2.showprintui#microsoft-web-webview2-core-corewebview2-showprintui)
+* [CoreWebView2.ShowPrintUI(CoreWebView2PrintDialogKind) Method](/dotnet/api/microsoft.web.webview2.core.corewebview2.showprintui#microsoft-web-webview2-core-corewebview2-showprintui(microsoft-web-webview2-core-corewebview2printdialogkind))
```

Comment:
```suggestion
* [CoreWebView2.ShowPrintUI(printDialogKind) Method](/dotnet/api/microsoft.web.webview2.core.corewebview2.showprintui#microsoft-web-webview2-core-corewebview2-showprintui(microsoft-web-webview2-core-corewebview2printdialogkind))
```
we've used this convention in past - when we (very rarely) needed to show params b/c an overloaded method: show the intent-driven param name rather than the verbose, technical type.

No overloads found at:
https://learn.microsoft.com/en-us/microsoft-edge/webview2/concepts/overview-features-apis?tabs=dotnetcsharp#web-messaging
ie Find of ") method" gives 0 hits.

RelNotes page might have some examples of our past recent conventions we developed:
found the one and only example of rare overload, and I plan to make this lowercase/camelcased like the WinRT tab:
https://learn.microsoft.com/en-us/microsoft-edge/webview2/release-notes?tabs=dotnetcsharp#experimental-features-3

Reply:
So, I updated the param name and also included two links here. One with the ShowPrintUI and other one with ShowPrintUI(printDialogKind) for dotnet. Can we have two here or just one with the overload?

* [CoreWebView2.ShowPrintUI Method](/dotnet/api/microsoft.web.webview2.core.corewebview2.showprintui#microsoft-web-webview2-core-corewebview2-showprintui)
* [CoreWebView2.ShowPrintUI(printDialogKind) Method](/dotnet/api/microsoft.web.webview2.core.corewebview2.showprintui#microsoft-web-webview2-core-corewebview2-showprintui(microsoft-web-webview2-core-corewebview2printdialogkind))